log/src/main/java/kcl/teamIndexZero/traffic/log/FileIO/FileOutput.java

```diff
@@ -83,52 +89,80 @@ public boolean closeWriter() {
 
     /**
      * Writes a string to the end of the file
+     *
      * @param string String to append to file
-     * @exception IOException when the file cannot be accessed
+     * @throws IOException when the file cannot be accessed
      */
-    public void appendString( String string ) throws InvalidPathException, IOException {
+    public synchronized void appendString(String string) throws InvalidPathException, IOException {
```

Comment:
... and `synchronized` here may make quite a lot of performance troubles to us. Consider a multi-threaded approach (say, we will have 3 threads computing different features of the mode) later. What will happen with this methods being synchronized, is essentially the threads would queue in series to write their messages to log file. 

I'd say, it is okay for now when we have single threaded app, but for the MT we'll need to be much more careful with synchro, moving towards that [BlockingQueue](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/BlockingQueue.html) or other Java MT primitive.